### PR TITLE
Skil 196

### PR DESCRIPTION
**Pull request by brianlugo1**
This PR includes:
- Redesign changes to the Login Page!
- Redesign for the Setup New Password Pages!
- Fixed Login Jest Tests to correctly target MUI Components!
- Fixed Jest error when trying to transpile an imported library!
- Added media tags for supporting different views!
- Replacing Error Message html tags for MUI Alert Component!
- Minor changes made to different View Tables!
- Minor changes made to CustomDataTable Component!

UPDATE:
- Fixed issue with logger not being able to parse the datetime stap from a line that is apart of a single error context!

### CODE REVIEW

File: FrontEndReact/src/View/Login/SetNewPassword.js [hunk 1]
```diff
@@ -3,87 +3,277 @@ import 'bootstrap/dist/css/bootstrap.css';
...
+                    break;
+                case 'STRONG':
+                    result = [COLORS.STRONG, COLORS.STRONG, COLORS.STRONG, COLORS.STRONG];
+                    break;
+            }
+        
+            return result;
+        };
+
+        this.testingPasswordStrength = (password) => {
```
Comment (malloc-nbytes) [inline FrontEndReact/src/View/Login/SetNewPassword.js:97]: maybe typo: did you mean `testPasswordStrength`?

File: FrontEndReact/src/View/Login/SetNewPassword.js [hunk 2]
```diff
@@ -3,87 +3,277 @@ import 'bootstrap/dist/css/bootstrap.css';
...
 class SetNewPassword extends Component {
     constructor(props) {
         super(props);
+
         this.state = {
             errorMessage: null,
-            passSet: false
+            passSet: false,
+            password: '',
+            password2: '',
```
Comment (malloc-nbytes) [inline FrontEndReact/src/View/Login/SetNewPassword.js:20]: I suggest renaming `password2` -> `confirmationPassword` or something similar.

File: FrontEndReact/src/View/Login/SetNewPassword.js [hunk 3]
```diff
@@ -3,87 +3,277 @@ import 'bootstrap/dist/css/bootstrap.css';
...
-            var pass2 = document.getElementById("password2").value;
-    
-            if (pass1 === pass2) {
+            var pass1 = this.state.password;
+            var pass2 = this.state.password2;
 
+            var passwordSecurity = this.testingPasswordStrength(pass1);
+
+            if (pass1 === pass2 && passwordSecurity === "STRONG") {
                 var valid = validPasword(pass1);
```
Comment (malloc-nbytes) [inline FrontEndReact/src/View/Login/SetNewPassword.js:128]: Is this line needed anymore?

File: FrontEndReact/src/View/Login/SetNewPassword.js [hunk 4]
```diff
@@ -3,87 +3,277 @@ import 'bootstrap/dist/css/bootstrap.css';
...
+            } else {
                 this.setState(() => ({
-                    errorMessage: "Passwords do not match"
-                }))
-            }
+                    errorMessage: "Passwords do not match",
 
+                    errors: {
+                        password: "Password do not match",
+                        password2: "Password do not match"
```
Comment (malloc-nbytes) [inline FrontEndReact/src/View/Login/SetNewPassword.js:149]: typo: Password -> Passwords

**Review by malloc-nbytes — CHANGES_REQUESTED** [review]
Only 2 big issues that need fixing (the two switch statements), as well as a few typos.

File: FrontEndReact/src/View/Admin/View/ViewAssessmentTask/ViewAssessmentTasks.js [hunk 5]
```diff
@@ -53,9 +53,9 @@ class ViewAssessmentTasks extends Component {
                         const monthNames = ["Jan","Feb","Mar","Apr","May","Jun","Jul","Aug","Sep","Oct","Nov","Dec"];
                         var due_date_string = `${monthNames[month]} ${(day)} at ${hour%12}:${minute<10?("0"+minute):minute}${hour<12?"am":"pm"}`;
```
[the remark this answers is not in the document]
Comment (brianlugo1) [reply FrontEndReact/src/View/Admin/View/ViewAssessmentTask/ViewAssessmentTasks.js]: I committed changes to refactor due_date_string by using individual variables!

File: FrontEndReact/src/View/Admin/View/ViewAssessmentTask/ViewAssessmentTasks.js [hunk 6]
```diff
@@ -101,9 +101,9 @@ class ViewAssessmentTasks extends Component {
                     setCellProps: () => { return { width:"100px"} },
                     customBodyRender: (ratings) => {
                         return(
-                            <p>
+                            <>
                                 {ratings ? (ratings ? "Yes" : "No") : "No"}
```
[the remark this answers is not in the document]
Comment (brianlugo1) [reply FrontEndReact/src/View/Admin/View/ViewAssessmentTask/ViewAssessmentTasks.js]: I committed changes to just have `ratings ? “Yes” : “No”`!

File: FrontEndReact/src/View/Admin/View/ViewAssessmentTask/ViewAssessmentTasks.js [hunk 7]
```diff
@@ -133,9 +133,9 @@ class ViewAssessmentTasks extends Component {
                     setCellProps: () => { return { width:"155px"} },
                     customBodyRender: (unit_of_assessment) => {
                         return(
-                            <p>
+                            <>
                                 {unit_of_assessment ? (unit_of_assessment ? "Yes" : "No") : "No"}
```
[the remark this answers is not in the document]
Comment (brianlugo1) [reply FrontEndReact/src/View/Admin/View/ViewAssessmentTask/ViewAssessmentTasks.js]: I just committed changes to implement `unit_of_assessment ? “Yes” : “No”`!

File: FrontEndReact/src/View/Error/ErrorMessage.js [hunk 8]
```diff
@@ -1,30 +1,37 @@
...
-                        (
-                            this.props.fetchedResource ?
-                            "Fetching " + this.props.fetchedResource :
-                            ( this.props.add ? "Creating a new " : "Updating a new ") + this.props.resource + " resulted in an error: "
-                        )
+                <Box sx={{ width: "100%", display: "flex", justifyContent: "center"}}>
+                    <Alert aria-label='error_message_alert' sx={{ width: "40%", mt: 2, position:"absolute" }} severity="error" variant="filled">
+                        {
+                            (
+                                this.props.fetchedResource ?
```
[the remark this answers is not in the document]
Comment (brianlugo1) [reply FrontEndReact/src/View/Error/ErrorMessage.js]: I committed changes that refactored the logic for Error Message Component!

File: FrontEndReact/src/View/Login/SetNewPassword.js [hunk 9]
```diff
@@ -3,87 +3,277 @@ import 'bootstrap/dist/css/bootstrap.css';
...
+
 
 class SetNewPassword extends Component {
     constructor(props) {
         super(props);
+
         this.state = {
             errorMessage: null,
-            passSet: false
+            passSet: false,
```
[the remark this answers is not in the document]
Comment (brianlugo1) [reply FrontEndReact/src/View/Login/SetNewPassword.js]: I committed changes for replacing passSet to isPasswordSet!

File: FrontEndReact/src/View/Login/SetNewPassword.js [hunk 10]
```diff
@@ -3,87 +3,277 @@ import 'bootstrap/dist/css/bootstrap.css';
...
+        this.generateColors = (strength) => {
+            let result = [];
+
+            const COLORS = {
+                NEUTRAL: '#E2E2E2',
+                WEAK: '#B40314',
+                MEDIUM: '#D39323',
+                STRONG: '#7B927F',
+            };
+
```
[the remark this answers is not in the document]
Comment (brianlugo1) [reply FrontEndReact/src/View/Login/SetNewPassword.js:71]: I committed changes to implement the switch logic in getIcon() example you provided!

File: FrontEndReact/src/View/Login/SetNewPassword.js (hunk 1 above)
Comment (brianlugo1) [reply FrontEndReact/src/View/Login/SetNewPassword.js]: I committed changes to fix the typo!

File: FrontEndReact/src/View/Login/SetNewPassword.js (hunk 2 above)
Comment (brianlugo1) [reply FrontEndReact/src/View/Login/SetNewPassword.js]: I committed changes to changed password2 to confirmationPassword!

File: FrontEndReact/src/View/Login/SetNewPassword.js (hunk 3 above)
Comment (brianlugo1) [reply FrontEndReact/src/View/Login/SetNewPassword.js]: I committed changes to remove unnecessary variable valid!

File: FrontEndReact/src/View/Login/SetNewPassword.js (hunk 4 above)
Comment (brianlugo1) [reply FrontEndReact/src/View/Login/SetNewPassword.js]: I committed changes to fix the typo!

**Comment by brianlugo1** [comment]
@malloc-nbytes I have addressed and resolved all suggestions!

**Review by jdkingsbury — APPROVED** [review]
LGTM

File: BackEndFlask/models/logger.py [hunk 11]
```diff
@@ -49,17 +49,32 @@ def __try_clear(self):
...
         now = datetime.now()
+
         for handler in self.logger.handlers:
             if isinstance(handler, logging.FileHandler):
                 with open(handler.baseFilename, 'r+') as f:
                     lines = f.readlines()
+
                     f.seek(0)
+
+                    last_parsed_time: None
```
Comment (malloc-nbytes) [inline BackEndFlask/models/logger.py:60]: `last_parsed_time: None` -> `last_parsed_time = None`

File: FrontEndReact/src/View/Admin/View/ViewAssessmentTask/ViewAssessmentTasks.js [hunk 12]
```diff
@@ -46,16 +46,25 @@ class ViewAssessmentTasks extends Component {
...
+
                         const monthNames = ["Jan","Feb","Mar","Apr","May","Jun","Jul","Aug","Sep","Oct","Nov","Dec"];
-                        var due_date_string = `${monthNames[month]} ${(day)} at ${hour%12}:${minute<10?("0"+minute):minute}${hour<12?"am":"pm"}`;
+
+                        var minutesString = minute < 10 ? ("0" + minute): minute;
+                        var twelveHourClock = hour < 12 ? "am": "pm";
+
+                        var timeString = `${hour % 12}:${minutesString}${twelveHourClock}`;
+
+                        var due_date_string = `${monthNames[month]} ${day} at ${timeString}`;
```
Comment (malloc-nbytes) [inline FrontEndReact/src/View/Admin/View/ViewAssessmentTask/ViewAssessmentTasks.js:62]: A lot of variables are using camelCase naming, but other variables use snake_case. I suggest making it uniform throughout all of the front-end.

File: FrontEndReact/src/View/Admin/View/ViewAssessmentTask/ViewAssessmentTasks.js [hunk 13]
```diff
@@ -117,9 +126,9 @@ class ViewAssessmentTasks extends Component {
                     setCellProps: () => { return { width:"20px"} },
                     customBodyRender: (suggestions) => {
                         return(
-                            <p>
+                            <>
                                 {suggestions ? (suggestions ? "Yes" : "No") : "No"}
```
Comment (malloc-nbytes) [inline FrontEndReact/src/View/Admin/View/ViewAssessmentTask/ViewAssessmentTasks.js:130]: `{suggestions ? (suggestions ? "Yes" : "No") : "No"}` -> `suggestions ? "Yes" : "No";`

File: FrontEndReact/src/View/Login/SetNewPassword.js [hunk 14]
```diff
@@ -3,87 +3,264 @@ import 'bootstrap/dist/css/bootstrap.css';
...
-            passSet
+            isPasswordSet,
+            password,
+            confirmationPassword,
+            errors
         } = this.state;
+       
+        const passwordStrength = this.testPasswordStrength(password)
+        const colors = this.generateColors(passwordStrength);
+        const Icon = this.getIcon(passwordStrength);
```
Comment (malloc-nbytes) [inline FrontEndReact/src/View/Login/SetNewPassword.js:154]: I suggest renaming `Icon` -> `icon`. Uppercase names are usually reserved for ADTs, classes, etc.

**Review by malloc-nbytes — APPROVED** [review]
Just a few renaming things. Other than that, LGTM!

File: BackEndFlask/models/logger.py (hunk 11 above)
Comment (brianlugo1) [reply BackEndFlask/models/logger.py]: I committed changes to replace the : for a =!

File: FrontEndReact/src/View/Admin/View/ViewAssessmentTask/ViewAssessmentTasks.js (hunk 13 above)
Comment (brianlugo1) [reply FrontEndReact/src/View/Admin/View/ViewAssessmentTask/ViewAssessmentTasks.js]: I committed changes to remove the outer ternary operator!

File: FrontEndReact/src/View/Login/SetNewPassword.js (hunk 14 above)
Comment (brianlugo1) [reply FrontEndReact/src/View/Login/SetNewPassword.js:154]: In this instance Icon is storing a component corresponding to the password strength. Therefore, I am going to leave the capitalized Icon variable name!

**Comment by brianlugo1** [comment]
@sah0017 This PR has received one approved code review! I am going to go ahead and merge this PR!

**Pull request merged**